File: csharp/Calculator.cs
using System;
class Calculator {
    static void Main() {
        Console.Write("Enter expression: ");
        string[] input = Console.ReadLine().Split(' ');
        double a = double.Parse(input[0]);
        char op = char.Parse(input[1]);
        double b = double.Parse(input[2]);
        double result = op switch {
            '+' => a + b,
            '-' => a - b,
            '*' => a * b,
            '/' => b != 0 ? a / b : 0,
            _ => 0
        };
        Console.WriteLine(result);
    }
}

File: rust/calculator.rs
use std::io;
fn main() {
    let mut input = String::new();
    println!("Enter expression: ");
    io::stdin().read_line(&mut input).unwrap();
    let parts: Vec<&str> = input.split_whitespace().collect();
    let a: f64 = parts[0].parse().unwrap();
    let op = parts[1];
    let b: f64 = parts[2].parse().unwrap();
    let result = match op {
        "+" => a + b,
        "-" => a - b,
        "*" => a * b,
        "/" => if b != 0.0 { a / b } else { 0.0 },
        _ => 0.0,
    };
    println!("{}", result);
}
